### DISCUSSION TITLE
No device availability in config mqtt topic

### DISCUSSION THREAD

**Discussion by michal-bartak** [Koenkk/zigbee2mqtt#12569] [General]
After dismounting Christmas trees (yeah, I know it was months ago) I found that some Zigbee devices remain turned on in Zigbee2MQTT.
After a moment of going through MQTT topics I found this:

```
zigbee2mqtt/zb_switch1/availability = false
```
but also:
```
homeassistant/switch/0x00158d000449dccc/switch/config = 
{"availability":[{"topic":"zigbee2mqtt/bridge/state"}],"command_topic":"zigbee2mqtt/zb_switch1/set", ....
```

Why device availability is not included in overall entity availability?
I remember from the past, it was possible. 
I found no option in the z2m config to change it. how can I achieve that?

**Comment by Koenkk** [comment]
Availability has always been published to a different topic, see https://github.com/Koenkk/zigbee2mqtt/issues/10802#issuecomment-1023525378 for the rationale

**Reply by michal-bartak** [reply]
Thank you. So I can see the change appeared this year.
But it seems it somehow affecting HA discovery.

I have `Legacy availability payload` still set to true (actually never touched it). But for some reason mentioned entity is not recognized by HA as unavailable.

Is there something I should do to reflect availability into HA entities?
Does HA support the JSON format for availability?
I have Legacy API set to true too. Does it somehow affect availability? Can I set it to false safely considering I have  HA 2021.5?

BTW layout of the settings GUI is somehow distorted:
![image](https://user-images.githubusercontent.com/7868445/169849533-813640e2-e948-430a-b8d9-64e346d65790.png)

**Reply by Koenkk** [reply]
> Does HA support the JSON format for availability?

Recent versions of Home Assistant do support this (2022.5.x should be good for sure)

**Reply by michal-bartak** [reply]
I changed those settings, and yes, both availabilities are all together in json and HA recognizes unavailability.
Great that it was applied instantly without the need to a restart
Thank you!

**Comment by michal-bartak** [comment]
Hmm. 
But something went wrong with other device the Hue strip, which was disconnected from AC for some time
I connected it, I can control it using Z2m but HA says it's unavailable

Here are data from mqtt:
```
{"availability":[{"topic":"zigbee2mqtt/bridge/state","value_template":"{{ value_json.state }}"},{"topic":"zigbee2mqtt/hue_strip_1/availability","value_template":"{{ value_json.state }}"}],"availability_mode":"all","brightness":true,"brightness_scale":254,"color_mode":true,"command_topic":"zigbee2mqtt/hue_strip_1/set","device":{"identifiers":["zigbee2mqtt_0x00178801049a89d8"],"manufacturer":"Philips","model":"Hue white and color ambiance LightStrip plus (915005106701)","name":"hue_strip_1","sw_version":"1.88.1"},"effect":true,"effect_list":["blink","breathe","okay","channel_change","finish_effect","stop_effect"],"max_mireds":500,"min_mireds":153,"name":"hue_strip_1","schema":"json","state_topic":"zigbee2mqtt/hue_strip_1","supported_color_modes":["xy","color_temp"],"unique_id":"0x00178801049a89d8_light_zigbee2mqtt"}
```
their change resulting by z2m restart (previously I mentioned that restart wasn't needed to achieve unavailability for other device)
![image](https://user-images.githubusercontent.com/7868445/170110309-0c74aac1-25e2-49f1-8120-e3927cb61e29.png)

and availability topic:
![image](https://user-images.githubusercontent.com/7868445/170110543-9636f68f-09dc-4286-9fde-5dbd239d402e.png)

Edit:
The second Z2M restart has recovered the HA entity from the unavailable state. Are you aware of such behavior? TBH I recall some messages on the forum that discovery needs double restarts. But it was 2 years ago and about Shelly devices supported by some python script. Maybe it's a 'feature' of HA discovery?
